controllers/core/event_controller.go

```diff
@@ -47,7 +49,20 @@ const (
 	ReconcileEventsLatencyMetrics = "reconcile_events"
 	LabelOperation                = "operation"
 	LabelTime                     = "time"
+	LabelCacheHit                 = "cache_hit"
+	LabelCacheMiss                = "cache_miss"
+	LabelCacheSetErr              = "cache_set_error"
+	CacheHitMetricsValue          = "NODE_CACHED"
+	CacheMissMetricsValue         = "NODE_NOT_CACHED"
+	CacheErrMetricsValue          = "NODE_CACHE_ERROR"
 	EventRegardingKind            = "Node"
+	SGPPrefix                     = "SGP"
+	CNPrefix                      = "CN"
+
+	// customized configurations for BigCache
+	InstancesCacheTTL     = 30 * time.Minute // scaling < 1k nodes should be under 20 minutes
+	InstancesCacheShards  = 32               // must be power of 2
+	InstancesCacheMaxSize = 2                // in MB
```

Comment:
With your 500 node test, how big was cache? Just curious is 2MB is enough

Reply:
Cache size was for 500 and roughly calculate 2MB should be good for up to 20k instance IDs with custom networking enabled and 40k instance IDs without custom networking. 20k entries plus 30 minutes expiry setting, I think 2MB should be good for cases I can think of. If cache gets full and thus causes cache miss, missing records should be ok for controller to function properly. But I am ok to increase it to 4MB just for in case there are very high churning rate on nodes. I mean the size should be enough for "static" group of nodes, but bigger might be safer for very "dynamic" group of nodes.